src/unix/timer.c

```diff
@@ -71,7 +72,7 @@ int uv_timer_start(uv_timer_t* handle,
   if (uv__is_active(handle))
     uv_timer_stop(handle);
 
-  clamped_timeout = handle->loop->time + timeout;
+  clamped_timeout = (handle->timeout <= 0 ? handle->loop->time : handle->timeout) + timeout;
```

Comment:
Sorry, it's my bad.
The callback is executed after timeout calculated indeed.
But in my environment, the problem is that loop->time is not always accurate for repeating timers, so that after many times of timer triggering, the total timeouts are always delayed several seconds, this is quite weird.